### PR TITLE
Clean the code

### PR DESCRIPTION
**Pull request by LeifuChen**
## Description
This is the PR to resolve the following issues.
- Clean the unnecessary tx error in the `SmartMarginOnboard` component
- Add the TODO comment to the temporary solution of the prices and fix the non-serializable value issue
- Apply a cleaner solution to the truncation fix by removing the `break` statement

### CODE REVIEW

File: sdk/services/prices.ts [hunk 1]
```diff
@@ -187,11 +187,12 @@ export default class PricesService {
 		);
 		const prices = (response ? Object.values(response).flat() : []) as SynthPrice[];
 
+		//TODO: remove this once the rates are fixed
 		return prices.map((price) =>
 			price.synth === MarketAssetByKey[FuturesMarketKey.sPEPEPERP]
-				? { ...price, rate: wei(price.rate.toString()).div(1e10) }
+				? { ...price, rate: wei(price.rate.toString()).div(1e10).toString() }
```
Comment (avclarke) [inline sdk/services/prices.ts:193]: Good catch looks like this is an important one to get in a hotfix

Comment (LeifuChen) [reply sdk/services/prices.ts:193]: Yes. Agree. The error is also very annoying in the dev.

File: utils/formatters/number.ts [hunk 2]
```diff
@@ -115,14 +115,12 @@ export const formatNumber = (value: WeiSource, options?: FormatNumberOptions) =>
...
-			if (weiValue.gt(threshold.value) && weiValue.gte(truncateThreshold)) {
-				truncation = threshold;
-				break;
-			}
-		}
-	}
+	truncation =
+		truncateThreshold && !truncation
+			? thresholds.find(
+					(threshold) => weiValue.gt(threshold.value) && weiValue.gte(truncateThreshold)
```
Comment (avclarke) [inline utils/formatters/number.ts:121]: Should this be `gte` in both places?

File: sdk/services/prices.ts (hunk 1 above)
Comment (avclarke) [reply sdk/services/prices.ts:193]: I will also just try and fix on the subgraph now

File: utils/formatters/number.ts (hunk 2 above)
Comment (LeifuChen) [reply utils/formatters/number.ts:121]: Yes. Correct. This should be `gte`. Otherwise, it will miss the threshold value. Will update this.

**Review by avclarke — APPROVED** [review]

**Review by platschi — APPROVED** [review]

**Pull request merged**